src/Extension/MemberResetExtension.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace SilverStripe\MFA\Extension;
+
+use SilverStripe\Forms\FieldList;
+use SilverStripe\ORM\DataExtension;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\FieldType\DBDatetime;
+use SilverStripe\ORM\ValidationException;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\PasswordEncryptor_NotFoundException;
+use SilverStripe\Security\RandomGenerator;
+
+/**
+ * Provides DB columns / methods for account resets on Members
+ *
+ * @package SilverStripe\MFA\Extension
+ * @property Member owner
+ */
+class MemberResetExtension extends DataExtension
+{
+    private static $db = [
+        'AccountResetHash' => 'Varchar(160)',
+        'AccountResetExpired' => 'Datetime',
+    ];
+
+    public function updateCMSFields(FieldList $fields)
+    {
+        $fields->removeByName(['AccountResetHash', 'AccountResetExpired']);
+
+        return $fields;
+    }
+
+    /**
+     * Mirrors the implementation in Member::generateAutologinTokenAndStoreHash(),
+     * but against the AccountReset fields.
+     *
+     * @return string
+     * @throws PasswordEncryptor_NotFoundException
+     * @throws ValidationException
+     */
+    public function generateAccountResetTokenAndStoreHash(): string
+    {
+        $lifetime = $this->owner->config()->auto_login_token_lifetime;
+
+        do {
+            $generator = new RandomGenerator();
```

Comment:
Do you need to instantiate a new instance for every iteration, or could this line be moved outside the loop?

Reply:
This was pulled verbatim from core, so I assumed 'yes', but looking at the implementation of `RandomGenerator` now I don't think it's necessary. I've moved it out.